widgets/hwmontemp_linux.lua

```diff
@@ -0,0 +1,60 @@
+----------------------------------------------------------------
+--   Licensed under the GNU General Public License v2
+--   (C) 2019, Alexander Koch <lynix47@gmail.com>
+----------------------------------------------------------------
+
+-- environment
+local io = { popen = io.popen, open = io.open }
+local assert = assert
+local setmetatable = setmetatable
+
+-- sysfs prefix for hwmon devices
+local sys_hwmon = "/sys/class/hwmon/"
+-- cache table for hwmon device names
+local paths = {}
+
+-- transparently caching hwmon device name lookup
+function name_to_path(name)
+    if paths[name] then return paths[name] end
+
+    for sensor in io.popen("ls -1 " .. sys_hwmon):lines() do
+        local path = sys_hwmon .. sensor
+        local f = assert(io.open(path .. "/name", "r"))
+        local sname = f:read("*line")
+        f:close()
+        if sname == name then
+            paths[name] = path
+            return path
+        end
+    end
+
+    return nil
+end
+
+-- hwmontemp: provides name-indexed temps from /sys/class/hwmon
+-- vicious.widgets.hwmontemp
+local hwmontemp_linux = {}
+
+function worker(format, warg)
+    assert(type(warg) == "table", "invalid hwmontemp argument: must be a table")
+    name = warg[1]
+
+    if not warg[2] then
+        input = 1
+    else
+        input = warg[2]
+    end
+
+    local sensor = name_to_path(name)
```

Comment:
Well, it _could_. But `thermal_linux` does not offer input selection for the `hwmon` zone, and name caching would not fit in there so nicely either as this only makes sense for `hwmon` and not `core` or `proc`.

It's up to you. If you don't like the approach of having this new widget then please feel free to close this pull request. I don't know if I'd be coming up with another one that extends `thermal_linux` in this case, depends on the amount spare-time I have.

Reply:
Would `vicious.widgets.thermal{'radeon', 'hwmon', 'temp1_input'}` give the desired result?

Reply:
well it would be still possible to keep backwards compatibility by still accepting the old format, but removing it from documentation and then just check in lua if the string starts with `/`.

Reply:
> (...) it'd be a lot better if we can have similar behaviour to the FreeBSD variant (input of absolute path to the file only (...)

That's _exactly_ what I do not want, as paths do change randomly in case of hwmon sensors. They depend on the device number, which in turn depends on kernel module load order. This is not deterministic, and the only reason why I have written my new widget (as stated on top).